src/participant/prompts/promptBase.ts

```diff
@@ -0,0 +1,92 @@
+import * as vscode from 'vscode';
+import type { ChatResult, ParticipantResponseType } from '../constants';
+
+export interface PromptArgsBase {
+  request: {
+    prompt: string;
+  };
+  context: vscode.ChatContext;
+}
+
+export abstract class PromptBase<TArgs extends PromptArgsBase> {
+  protected abstract getAssistantPrompt(args: TArgs): string;
+
+  protected abstract getUserPrompt(args: TArgs): Promise<string>;
+
+  async buildMessages(args: TArgs): Promise<vscode.LanguageModelChatMessage[]> {
+    return [
+      // eslint-disable-next-line new-cap
+      vscode.LanguageModelChatMessage.Assistant(this.getAssistantPrompt(args)),
+      ...this.getHistoryMessages({ context: args.context }),
+      // eslint-disable-next-line new-cap
+      vscode.LanguageModelChatMessage.User(await this.getUserPrompt(args)),
+    ];
+  }
+
+  // When passing the history to the model we only want contextual messages
+  // to be passed. This function parses through the history and returns
+  // the messages that are valuable to keep.
+  // eslint-disable-next-line complexity
+  protected getHistoryMessages({
+    connectionNames,
+    context,
+  }: {
+    connectionNames?: string[]; // Used to scrape the connecting messages from the history.
```

Comment:
```suggestion
    connectionNames: string[]; // Used to scrape the connecting messages from the history.
```
[nit] Are there any cases where we don't send this? Maybe we can make the type stricter.